mteb/tasks/Retrieval/multilingual/StatcanDialogueDatasetRetrieval.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import MultilingualTask
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+_EVAL_SPLITS = ["dev", "test"]
+
+_LANGS = {
+    # <iso_639_3>-<ISO_15924>
+    "english": ["eng-Latn"],
+    "french": ["fra-Latn"],
+}
+
+
+def _load_statcan_data(
+    path: str, langs: list, splits: str, cache_dir: str = None, revision: str = None
+):
+    queries = {lang: {split: {} for split in splits} for lang in langs}
+    corpus = {lang: {split: {} for split in splits} for lang in langs}
+    relevant_docs = {lang: {split: {} for split in splits} for lang in langs}
+
+    for split in splits:
+        for lang in langs:
+            query_table = datasets.load_dataset(
+                path,
+                f"queries_{lang}",
+                split=split,
+                cache_dir=cache_dir,
+                revision=revision,
+            )
+            corpus_table = datasets.load_dataset(
+                path,
+                "corpus",
+                split=lang,
+                cache_dir=cache_dir,
+                revision=revision,
+            )
+
+            for row in query_table:
+                query = row["query"]
+                query_id = row["query_id"]
+                doc_id = row["doc_id"]
+                queries[lang][split][query_id] = query
```

Comment:
Current for conversational QA the expectation is that query is list of strings, however in StatsCan each query is a dictionary with "role" and "content" keys. This format is definitely better as it allows for dynamic number of utterances per turn and also supports multi-user-agent conversations. However some changes will be required to make it work.

1. The [check](https://github.com/embeddings-benchmark/mteb/blob/eea434adcce71cabf9ebfbecd111f929eba68434/mteb/evaluation/evaluators/RetrievalEvaluator.py#L79) that decides between `encode_queries` and `encode_conversations` will change.
2. The [default method for converting queries to input string](https://github.com/embeddings-benchmark/mteb/blob/eea434adcce71cabf9ebfbecd111f929eba68434/mteb/evaluation/evaluators/RetrievalEvaluator.py#L402) will also change.

Reply:
I can also change TopiOCQA loading accordingly once the above mentioned changes are done.

Reply:
Maybe this would be better as new issue?

Reply:
Changes to be made [here](https://github.com/embeddings-benchmark/mteb/blob/eea434adcce71cabf9ebfbecd111f929eba68434/mteb/evaluation/evaluators/RetrievalEvaluator.py#L402)